$NetBSD: patch-meson.build,v 1.2 2022/09/10 13:47:01 wiz Exp $

Avoid libdl dependency on NetBSD.
https://gitlab.gnome.org/GNOME/gimp/-/issues/8604

Fix Linux Input detection, using patch from upstream.
https://gitlab.gnome.org/GNOME/gimp/-/issues/8605

--- meson.build.orig	2022-08-21 19:21:38.000000000 +0000
+++ meson.build
@@ -127,6 +127,10 @@ platform_linux = (
   host_os.contains('linux')
 )
 
+platform_netbsd = (
+  host_os.contains('netbsd')
+)
+
 platform_windows = (
   host_os.contains('mingw') or
   host_os.contains('cygwin') or
@@ -321,7 +325,7 @@ conf.set('ENABLE_RELOCATABLE_RESOURCES',
 
 
 math              = cc.find_library('m')
-dl                = platform_windows ? no_dep : cc.find_library('dl')
+dl                = (platform_windows or platform_netbsd) ? no_dep : cc.find_library('dl')
 rpc               = platform_windows ? cc.find_library('rpcrt4') : no_dep
 dbghelp           = platform_windows ? cc.find_library('dbghelp') : no_dep
 winsock           = platform_windows ? cc.find_library('ws2_32') : no_dep
@@ -845,15 +849,23 @@ endif
 alsa = dependency('alsa', version: '>=1.0.0', required: get_option('alsa'))
 conf.set('HAVE_ALSA', alsa.found())
 
+# Linux Input
 
 if get_option('linux-input').disabled()
   have_linuxinput = false
-elif get_option('linux-input').enabled() and not cc.has_header('linux/input.h')
-  error('linux/input.h header not found.')
 else
-  have_linuxinput = x11_target
+  have_linuxinput = cc.has_header('linux/input.h',
+                                  required: get_option('linux-input'))
 endif
 
+if have_linuxinput
+  gudev = dependency('gudev-1.0', version: '>=167', required: get_option('gudev'))
+else
+  gudev = no_dep
+endif
+conf.set('HAVE_LIBGUDEV', gudev.found())
+
+
 # DirectX DirectInput
 directx = no_dep
 directx_sdk_path = get_option('directx-sdk-dir')
@@ -874,9 +886,6 @@ if directx_sdk_path != '' and platform_w
 endif
 conf.set('HAVE_DX_DINPUT', directx.found())
 
-gudev = dependency('gudev-1.0', version: '>=167', required: get_option('gudev'))
-conf.set('HAVE_LIBGUDEV', gudev.found())
-
 
 ################################################################################
 # Email sending
